src/main/java/com/yelp/nrtsearch/server/luceneserver/IndexState.java

```diff
@@ -530,6 +541,9 @@ public synchronized void load(JsonObject jsonObject) throws IOException {
   /** The field definitions (registerField) */
   private final Map<String, FieldDef> fields = new ConcurrentHashMap<String, FieldDef>();
 
+  /** The meta field definitions */
+  public static Map<String, FieldDef> metaFields;
```

Comment:
It would be good to change IndexState::PerFieldSimilarityWrapper to remove `if (fields.containsKey(name))`. It would also be good to have IndexState.addField() reject fields with the same name as a meta field.